packages/circuit-ui/components/SidePanel/SidePanel.stories.tsx

```diff
@@ -0,0 +1,239 @@
+            </Button>
+          )}
+          <Button
+            type="button"
+            variant="tertiary"
+            size="kilo"
+            onClick={onClose}
+            css={spacing({ left: 'mega', bottom: 'mega' })}
+          >
+            Close
+          </Button>
+        </>
+      )}
+    </>
+  );
+};
+
+const ComponentWithSidePanel = (props) => {
+  const [selectedItem, setSelectedItem] = useState<string>(null);
+  const { setSidePanel } = useSidePanel();
+
+  return (
+    <ListItemGroup
+      items={items.map((item) => ({
+        ...item,
+        variant: 'navigation',
+        selected: item.key === selectedItem,
+        onClick: () => {
+          setSelectedItem(item.key);
+          setSidePanel({
+            ...props,
+            children: <DefaultChildren label={item.label} showMoreInfo />,
+          });
+        },
+      }))}
+      label="Select an item to open its details in a side panel"
+      css={spacing('mega')}
+    />
+  );
+};
+
+export const Base = (props: SidePanelHookProps): JSX.Element => (
+  <SidePanelProvider>
+    <ComponentWithSidePanel {...props} />
+  </SidePanelProvider>
+);
+Base.args = baseArgs;
+
+export const WithTopNavigation = (props: SidePanelHookProps): JSX.Element => (
+  <>
+    <TopNavigation {...topNavigationProps} />
+    <SidePanelProvider withTopNavigation>
+      <ComponentWithSidePanel {...props} />
+    </SidePanelProvider>
+  </>
+);
+WithTopNavigation.args = baseArgs;
+
+const ComponentWithSidePanelExtended = (props) => {
+  const [selectedItem, setSelectedItem] = useState<string>(null);
+  const { setSidePanel, updateSidePanel, removeSidePanel } = useSidePanel();
+
+  return (
+    <ListItemGroup
+      items={items.map((item) => ({
+        ...item,
+        variant: 'navigation',
+        selected: item.key === selectedItem,
+        onClick: () => {
+          if (selectedItem === item.key) {
+            setSelectedItem(null);
+            removeSidePanel();
+          } else {
+            setSidePanel({
+              ...props,
+              children: ({ onClose }) => (
+                <DefaultChildren
+                  label={item.label}
+                  showLoading
+                  onClose={onClose}
+                />
+              ),
+              onClose: () => setSelectedItem(null),
+            });
+            setSelectedItem(item.key);
+            setTimeout(() => {
+              updateSidePanel({
+                children: ({ onClose }) => (
+                  <DefaultChildren
+                    label={item.label}
+                    showMoreInfo
+                    showClose
+                    onClose={onClose}
+                  />
+                ),
+              });
+            }, 1000);
+          }
+        },
+      }))}
+      label="Select an item to open its details in a side panel"
+      css={spacing('mega')}
+    />
+  );
+};
+
+export const UpdateAndRemove = (props: SidePanelHookProps): JSX.Element => (
```

Comment:
Oh, that's really sad. It's a styling problem with the `ListItem` when the component returns the focus to the trigger element, but it's not `:focus-visible` in order to also show the blue outline. Unfortunately, if I remember correctly it was some pretty messy CSS. I'll look into it.

Reply:
In the end I couldn't think of a CSS-only solution because of the `li` wrapper around `ListItem` :( I had to add `onFocus` and `onBlur` handlers and check for `:focus-visible` if it is supported by the browser.